mobile/src/main/java/com/gelakinetic/mtgfam/helpers/GraphHelper.java

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright 2017 Adam Feinstein
+ *
+ * This file is part of MTG Familiar.
+ *
+ * MTG Familiar is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MTG Familiar is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MTG Familiar.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.gelakinetic.mtgfam.helpers;
+
+import android.content.Context;
+
+import com.gelakinetic.mtgfam.R;
+import com.gelakinetic.mtgfam.helpers.view.ReliableColorPie;
+import com.github.mikephil.charting.charts.BarChart;
+import com.github.mikephil.charting.charts.PieChart;
+import com.github.mikephil.charting.components.AxisBase;
+import com.github.mikephil.charting.components.Legend;
+import com.github.mikephil.charting.components.LegendEntry;
+import com.github.mikephil.charting.components.XAxis;
+import com.github.mikephil.charting.data.BarData;
+import com.github.mikephil.charting.data.BarDataSet;
+import com.github.mikephil.charting.data.BarEntry;
+import com.github.mikephil.charting.data.PieData;
+import com.github.mikephil.charting.data.PieDataSet;
+import com.github.mikephil.charting.data.PieEntry;
+import com.github.mikephil.charting.formatter.ValueFormatter;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class GraphHelper {
+    private final Context c;
+    private final DeckStatsGenerator mStatGenerator;
+    private final int mBackgroundColor;
+    private ArrayList<String> mTypes;
+
+    /**
+     * Creates a new GraphHelper
+     * @param mStatGenerator DeckStatsGenerator to get stats from
+     * @param c Context to get colors with
+     */
+    public GraphHelper(DeckStatsGenerator mStatGenerator, Context c) {
+        this.mStatGenerator = mStatGenerator;
+        this.c = c;
+        mBackgroundColor = c.getResources().getColor(R.color.background_light);
+    }
+
+    /**
+     * Fills all of the stat graphs at once, with the added benefit of adding an invisible legend
+     * to equally balance the color graph with the type graph
+     * @param chartToFill type PieChart to fill
+     * @param pieToFill color ReliableColorPie to fill
+     * @param barChartToFill cmc BarChart to fill
+     */
+    public void fillStatGraphs(PieChart chartToFill, ReliableColorPie pieToFill, BarChart barChartToFill) {
+        fillTypeGraph(chartToFill);
+        fillCmcGraph(barChartToFill);
+        fillColorGraph(pieToFill);
+        Legend pieLegend = pieToFill.getLegend();
```

Comment:
Filling out the legend with type values should be moved into `fillTypeGraph()`

Reply:
And then calling `chartToFill.setDrawEntryLabels(true);` instead of `chartToFill.setDrawEntryLabels(false);` draws the labels on the wedges nicely.